pallets/dactr/src/benchmarking.rs

```diff
@@ -1,23 +1,169 @@
 #![cfg(feature = "runtime-benchmarks")]
 
-use frame_benchmarking::{benchmarks, whitelisted_caller};
-use frame_support::traits::Get;
-use frame_system::RawOrigin;
-use sp_std::{fmt::Debug, iter::repeat, vec::Vec};
+use codec::{Decode, Encode};
+use da_primitives::{
+	asdr::{AppExtrinsic, AppUncheckedExtrinsic},
+	BlockLengthColumns, BlockLengthRows, OpaqueExtrinsic, BLOCK_CHUNK_SIZE, NORMAL_DISPATCH_RATIO,
+};
+use frame_benchmarking::{benchmarks, vec, whitelisted_caller};
+use frame_support::{log::info, traits::Get};
+use frame_system::{
+	header_builder::hosted_header_builder, limits::BlockLength, submitted_data, RawOrigin,
+};
+use scale_info::{StaticTypeInfo, TypeInfo};
+use sp_core::H256;
+use sp_runtime::{
+	traits::{DispatchInfoOf, Dispatchable, SignedExtension},
+	transaction_validity::{TransactionValidity, TransactionValidityError},
+};
+use sp_std::{
+	fmt::Debug,
+	iter::{once, repeat},
+	vec::Vec,
+};
 
-use crate::*;
+use crate::{pallet::Call as DACall, *};
+
+type RuntimeCallOf<T> = <T as frame_system::Config>::RuntimeCall;
+
+#[derive(PartialEq, Eq, Clone, Debug, Encode, Decode, TypeInfo)]
+pub struct SignedExtensionUnused<
+	T: frame_system::Config + Send + Sync + pallet::Config + Debug + StaticTypeInfo,
+>(
+	pub  (
+		frame_system::CheckNonZeroSender<T>,
+		frame_system::CheckSpecVersion<T>,
+		frame_system::CheckTxVersion<T>,
+		frame_system::CheckGenesis<T>,
+		frame_system::CheckEra<T>,
+		frame_system::CheckNonce<T>,
+		frame_system::CheckWeight<T>,
+	),
+);
+
+impl<T: frame_system::Config + Send + Sync + pallet::Config + Debug + StaticTypeInfo>
+	SignedExtension for SignedExtensionUnused<T>
+where
+	T: frame_system::Config + Send + Sync + pallet::Config + Debug + StaticTypeInfo,
+	RuntimeCallOf<T>: Dispatchable<RuntimeOrigin = T::RuntimeOrigin> + From<DACall<T>>,
+{
+	type AccountId = T::AccountId;
+	type AdditionalSigned = ();
+	type Call = RuntimeCallOf<T>;
+	type Pre = ();
+
+	const IDENTIFIER: &'static str = "SignedExtensionUnused";
+
+	fn additional_signed(&self) -> Result<Self::AdditionalSigned, TransactionValidityError> {
+		Ok(())
+	}
+
+	fn pre_dispatch(
+		self,
+		_who: &Self::AccountId,
+		_call: &Self::Call,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
+	) -> Result<(), TransactionValidityError> {
+		Ok(())
+	}
+
+	fn validate(
+		&self,
+		_who: &Self::AccountId,
+		_call: &Self::Call,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
+	) -> TransactionValidity {
+		Ok(Default::default())
+	}
+}
+
+fn repeat_bytes(byte: u8, len: u32) -> Vec<u8> {
+	repeat(byte).take(len as usize).collect::<Vec<_>>()
+}
+
+fn prev_power_of_two(n: u32) -> u32 {
```

Comment:
Minor thing: Does it repeat several times the same power of 2? Kind of nested iteration if we thought about `--repeat=10` parameter when we run the benchmark.
Is there any way to fix also the rows to `[128,256,512,1024]` ? I guess benchmarks will be faster and we will see less deviation.

Reply:
The goal was to actually have meaningful repetitions. We can fix rows when using power of 2s for the range instead of plain rows. In the end I set it up like this for better comprehension, but I can revert it if you prefer. (I think since we define steps and repetitions, end result should be same)